File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        int n = A.size();
        vector <int> res(n);

        vector<int> freq(n+1,0);

        for(int i = 0 ; i<n; i++) {
            int count = 0;
            freq[A[i]]++;
            freq[B[i]]++;
            for(int j=0; j<=n; j++) {
                if(freq[j] == 2) {
                    count++;
                }
            }
             res[i] = count;
        }
        return res;
    }
};

File: 2766-find-the-prefix-common-array-of-two-arrays/find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        ans = []

        for i in range(n) :
            count = 0

            for num in range(1, n+1):
                if num in A[:i+1] and num in B[:i+1] :
                    count +=1

            ans.append(count)
        return ans
        